packages/mdc-list/index.js

```diff
@@ -80,27 +85,62 @@ export class MDCList extends MDCComponent {
 
   /** @return Array<!Element>*/
   get listElements_() {
-    return [].slice.call(this.root_.querySelectorAll(strings.ITEMS_SELECTOR))
-      .filter((ele) => ele.parentElement === this.root_);
+    return [].slice.call(this.root_.querySelectorAll(strings.ITEMS_SELECTOR));
   }
 
   /** @param {boolean} value */
   set wrapFocus(value) {
     this.foundation_.setWrapFocus(value);
   }
 
+  /** @param {boolean} isSingleSelectionList */
+  set singleSelection(isSingleSelectionList) {
+    if (isSingleSelectionList) {
+      this.root_.addEventListener('click', this.handleClick_);
+    } else {
+      this.root_.removeEventListener('click', this.handleClick_);
+    }
+
+    this.foundation_.setSingleSelection(isSingleSelectionList);
+    const selectedElement = this.root_.querySelector('.mdc-list-item--selected');
+
+    if (selectedElement) {
+      this.selectedIndex = this.listElements_.indexOf(selectedElement);
+    }
+  }
+
+  /** @param {number} index */
+  set selectedIndex(index) {
+    this.foundation_.setSelectedIndex(index);
+  }
+
   /** @return {!MDCListFoundation} */
   getDefaultFoundation() {
-    return new MDCListFoundation(/** @type {!MDCListAdapter} */{
+    return new MDCListFoundation(/** @type {!MDCListAdapter} */ (Object.assign({
       getListItemCount: () => this.listElements_.length,
       getFocusedElementIndex: () => this.listElements_.indexOf(document.activeElement),
       getListItemIndex: (node) => this.listElements_.indexOf(node),
-      focusItemAtIndex: (ndx) => this.listElements_[ndx].focus(),
+      setAttributeForElementIndex: (index, attr, value) => this.listElements_[index].setAttribute(attr, value),
+      removeAttributeForElementIndex: (index, attr) => this.listElements_[index].removeAttribute(attr),
+      addClassForElementIndex: (index, className) => this.listElements_[index].classList.add(className),
+      removeClassForElementIndex: (index, className) => this.listElements_[index].classList.remove(className),
```

Comment:
I checked the foundation and it looks like you have null check code there (eg. this.selectedIndex >= 0), but is there a possibility that `this.listElements_[index]` could be undefined? Just wanna check